rcl_yaml_param_parser/src/parse.c

```diff
@@ -809,18 +801,37 @@ rcutils_ret_t find_parameter(
       return RCUTILS_RET_OK;
     }
   }
-  // Parameter not found, add it.
   rcutils_allocator_t allocator = param_st->allocator;
+  // Reallocate if necessary
+  bool reallocate = false;
+  rcutils_ret_t ret;
+  if (node_param_st->num_params >= node_param_st->capacity_params) {
+    if (RCUTILS_RET_OK != node_params_reallocate(
+        node_param_st, allocator, node_param_st->capacity_params * 2))
+    {
+      ret = RCUTILS_RET_BAD_ALLOC;
+      goto clean;
+    }
+    reallocate = true;
+  }
+  // Parameter not found, add it.
```

Comment:
I think this comment belongs in the original place, that is, right after trying to find the parameter.